### PR TITLE
BUG: Added removal of meta attributes as claimed by docstring

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

Addresses #612 

Meta attributes weren't removed from .Meta when transferred to the Instrument object as claimed by the docstring.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Modified unit tests to check for removal of attributes.

**Test Configuration**:
* macOS 10.15.4
* python 2

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: pysat/_meta.py [hunk 1]
```diff
@@ -1155,30 +1157,28 @@ def transfer_attributes_to_instrument(self, inst, strict_names=False):
         for key in this_attrb:
             if key not in banned:
                 if key not in base_attrb:
-                    # don't store _ leading attributes
+                    # don't store private attributes
```
Comment (aburrell) [inline pysat/_meta.py:1160]:
```suggestion
                    # Don't store any hidden attributes
```

File: pysat/tests/test_meta.py [hunk 2]
```diff
@@ -1021,39 +1021,42 @@ def test_replace_meta_units_list_weird_case(self):
...
             self.meta.new_attribute = 'hello'
-            self.meta.__yo_yo = 'yo yo'
-            self.meta.date = None
             self.meta.transfer_attributes_to_instrument(self.testInst)
+
+            # test private not transferred
+            with pytest.raises(AttributeError):
+                self.testInst._yo_yo
             with pytest.raises(AttributeError):
                 self.testInst.__yo_yo
```
Comment (aburrell) [inline pysat/tests/test_meta.py:1052]:
```suggestion
            assert not hasattr(self.testInst, "_yo_yo")
            assert not hasattr(self.testInst, "__yo_yo")
```

Comment (rstoneback) [reply pysat/tests/test_meta.py:1052]: Thanks!

**Comment by rstoneback** [comment]
@aburrell ready for re-review when you are available.

**Review by aburrell — APPROVED** [review]

**Pull request merged**